pandas/tests/series/methods/test_interpolate.py

```diff
@@ -340,6 +340,14 @@ def test_interp_invalid_method(self, invalid_method):
         with pytest.raises(ValueError, match=msg):
             s.interpolate(method=invalid_method, limit=-1)
 
+    def test_interp_invalid_method_and_value(self):
+        # GH#36624
+        ser = Series([1, 3, np.nan, 12, np.nan, 25])
+
+        msg = "Cannot pass both fill_value and method"
+        with pytest.raises(ValueError, match=msg):
+            ser.interpolate(fill_value=3, method="pad")
```

Comment:
fill_value is not in the Series.interpolate api. This is accepted through kwargs from the public api? Then fill_value is accepted and passed along internally? I assume none of the scipy interpolate functions accept fill_value? or won't in the future!

do we really need the ValueError if fill_value is passed?

Reply:
> do we really need the ValueError if fill_value is passed?

In master:

```
>>> ser = pd.Series([1, 2, np.nan, 4])
>>> ser.interpolate(method="pad", fill_value=99)
0    1.0
1    2.0
2    NaN
3    4.0
dtype: float64
```

I'm pretty confident this isn't intentional.

Reply:
but on master and this PR
```
>>> pd.__version__
'1.2.0.dev0+499.g3b12293cea'
>>> ser = pd.Series([1, 2, np.nan, 4])
>>> ser.interpolate(fill_value=99)
0    1.0
1    2.0
2    3.0
3    4.0
dtype: float64

>>> ser.interpolate(humpty_dumpty=99)
0    1.0
1    2.0
2    3.0
3    4.0
dtype: float64
```
why the ValueError when method passed?

Reply:
1) it may be that we can disallow fill_value at the top of Series.interpolate.  making `**kwargs` explicit would be nice.  But i haven't thoroughly checked if thats possible.
2) BlockManager.interpolate also gets called from fillna, so we need to rule out fill_value at the place where this PR does it.

Reply:
this looks ok to me; i think we are accepting too much here accidently.

@jbrockmendel can you add a whatsnew note in any case (just saying passing invalid keyword args to `.interpolate()` will raise (or can do it after when all non-named keyword args are banned.

Reply:
so a ValueError may be appropriate for fill_na, but the test is for Series.interpolate. Why not `TypeError: interpolate() got an unexpected keyword argument 'fill_value'`?